Adjust "add" message for Coq 8.15

jfehrle opened the pull request:
~I'd like to merge https://github.com/coq/coq/pull/14252 for Coq 8.14.~  That PR modifies the `add` message, so the coqtail tests need to be updated.  The changes update the `add` message to add new values to the sent type and remove an unused value from the return type.  The added values are relevant only for the visual debugger.

I think my code for adding the new parameter is close to correct, but I couldn't find the code that checks for the value in the return type.  Can you tell me what needs to change for that?

~I'd like to get my Coq PR into 8.14, which has a freeze date of May 31.  I hope we can update the tests in the next few days rather than waiting until the last minute.~

Let me know your thoughts, thanks.

The message types change from:
```
type add_sty = (string * edit_id) * (state_id * verbose)
type add_rty = state_id * ((unit, state_id) union * string)
```

to:

```
type add_sty = (((string * edit_id) * (state_id * verbose)) * int) * (int * int)
type add_rty = state_id * (unit, state_id) union
```

Updated 14 Sept 2021  The associated Coq PR is now https://github.com/coq/coq/pull/14644

jfehrle commented:
> As far as testing, I'm not sure what the best approach is. Coqtail's CI runs against Coq master so this PR will fail that test until your Coq PR is merged.

I'll use Coq CI's "overlay" mechanism that permits running our tests using code from, for example, this PR.  I'll give that a whirl and let you know, thanks.

jfehrle commented:
The tests now pass for the Coq PR.  I'll let you know when the Coq change merges, at which time this should be merged.

jfehrle commented:
BTW, it appeared that the coqtail tests hang indefinitely when Coq and the tests disagree on the message types (e.g. without this overlay).  I think CI kills the job after 3 hours.  Perhaps the test code could someday apply a much shorter timeout, perhaps even less than a minute per test case.

jfehrle commented:
FYI: This won't be in Coq 8.14.  The review of coq/coq#14252 has been delayed and is still pending.

whonore commented:
Thanks for letting me know, I won't bother continuing to rebase it then until it's ready to be merged. It should be a straightforward fix at that point though.

jfehrle commented:
FYI: I added 2 more parameters to the `add` message and updated the description at the top of this page.

I hope the debugger makes in into Coq 8.15.  The 8.14 beta may soon finally appear.

jfehrle commented:
@whonore Looks we'll finally merge the debugger PR in Coq in the next few days.  I'm not sure how I ended up with 5 commits.  You may want to simplify that.  Thanks.

ejgallego commented:
Upstream changes have been merged, please merge.

A backwards-compatible alternative is to keep the actual type of add and add a new call `add_with_loc`
